src/spring-cloud/azext_spring_cloud/gateway.py

```diff
@@ -0,0 +1,232 @@
+                                 domain_name,
+                                 certificate=None):
+    properties = models.GatewayCustomDomainProperties()
+    if certificate is not None:
+        certificate_response = client.gateways.certificates.get(
+            resource_group, service, certificate)
+        properties = models.GatewayCustomDomainProperties(
+            thumbprint=certificate_response.properties.thumbprint
+        )
+
+    custom_domain_resource = models.GatewayCustomDomainResource(
+        properties=properties)
+    return client.gateway_custom_domains.begin_create_or_update(resource_group, service, DEFAULT_NAME,
+                                                                domain_name, custom_domain_resource)
+
+
+def gateway_custom_domain_unbind(cmd, client, resource_group, service, domain_name):
+    client.gateway_custom_domains.get(resource_group, service,
+                                      DEFAULT_NAME, domain_name)
+    return client.gateway_custom_domains.begin_delete(resource_group, service, DEFAULT_NAME, domain_name)
+
+
+def gateway_route_config_show(cmd, client, resource_group, service, name):
+    return client.gateway_route_configs.get(resource_group, service, DEFAULT_NAME, name)
+
+
+def gateway_route_config_list(cmd, client, resource_group, service):
+    return client.gateway_route_configs.list(resource_group, service, DEFAULT_NAME)
+
+
+def gateway_route_config_create(cmd, client, resource_group, service, name,
+                                app_name=None,
+                                routes_json=None,
+                                routes_file=None):
+    if routes_json is not None and routes_file is not None:
+        raise CLIError(
+            "You can only specify either --routes-json or --routes-file.")
+
+    route_configs = client.gateway_route_configs.list(
+        resource_group, service, DEFAULT_NAME)
+    if name in (route_config.name for route_config in list(route_configs)):
+        raise CLIError("Route config " + name + " already exists")
+
+    app_resource = client.apps.get(resource_group, service, app_name)
+
+    routes = []
+    if routes_file is not None:
+        with open(routes_file, 'r') as json_file:
+            routes = json.load(json_file)
+
+    if routes_json is not None:
+        routes = json.loads(routes_json)
+
+    properties = models.GatewayRouteConfigProperties(
+        app_resource_id=app_resource.id, routes=routes)
+
+    route_config_resource = models.GatewayRouteConfigResource(
+        properties=properties)
+    return client.gateway_route_configs.begin_create_or_update(resource_group, service, DEFAULT_NAME, name, route_config_resource)
+
+
+def gateway_route_config_update(cmd, client, resource_group, service, name,
+                                app_name=None,
+                                routes_json=None,
+                                routes_file=None):
+    if routes_json is not None and routes_file is not None:
```

Comment:
This method almost share the same logic as create. We can move the exist check to validator and then use the same method to serve the two command